src/plugins/charset_converter.c

```diff
@@ -0,0 +1,185 @@
+    struct std_value *args, *tmp;
+    char *charset, *data;
+
+    args = &object->std_arg;
+
+    if (args == NULL || !STDVALUE_IS_MAP(*args)) {
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg` to be a map.");
+    }
+
+    tmp = stdmap_get_str(&object->std_arg, "charset");
+    if (tmp == NULL || !STDVALUE_IS_STRING(*tmp)) {
+        LOG_ERROR("Call missing mandatory parameter charset.\n");
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg['charset'] to be a string.");
+    }
+
+    charset = STDVALUE_AS_STRING(*tmp);
+
+    tmp = stdmap_get_str(&object->std_arg, "data");
+    if (tmp == NULL || !STDVALUE_IS_STRING(*tmp)) {
+        LOG_ERROR("Call missing mandatory parameter data.\n");
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg['data'] to be a string.");
+    }
+
+    data = STDVALUE_AS_STRING(*tmp);
+
+    char* from = (char*) malloc(strlen(data) + 1);
+    char* to = (char*) malloc(strlen(data) + 1);
+    strcpy(from, data);
+
+    bool res = convert(from, to, strlen(from) + 1, "UTF-8", charset);
+    if(!res)
+    {
+        return platch_respond_error_std(responseHandle, "error_id", "charset_name_unrecognized", NULL);
+    }
+
+    uint8_t* output;
+    output = (uint8_t*)malloc(strlen(to));
+
+    for(int i = 0; i < strlen(to); i++)
+    {
+        output[i] = (uint8_t)to[i];
+    }
+
+    return platch_respond(
+        responseHandle,
+        &(struct platch_obj){ .codec = kStandardMethodCallResponse, .success = true, .std_result = { .type = kStdUInt8Array, .uint8array = &output[0], .size = strlen(to) }, }
+    );
+}
+
+static int on_available_charsets(struct platch_obj *object, FlutterPlatformMessageResponseHandle *responseHandle) {
+    (void) object;
+
+    struct std_value* list = (struct std_value[]){ { .type = kStdString, .string_value = "Not available for Linux." }};
+
+    return platch_respond(
+        responseHandle,
+        &(struct platch_obj){ .codec = kStandardMethodCallResponse, .success = true, .std_result = { .type = kStdList, .list = list, .size = 1 }, }
+    );
+}
+
+static int on_check(struct platch_obj *object, FlutterPlatformMessageResponseHandle *responseHandle) {
+    struct std_value *args, *tmp;
+    char *charset;
+
+    args = &object->std_arg;
+
+    if (args == NULL || !STDVALUE_IS_MAP(*args)) {
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg` to be a map.");
+    }
+
+    tmp = stdmap_get_str(&object->std_arg, "charset");
+    if (tmp == NULL || !STDVALUE_IS_STRING(*tmp)) {
+        LOG_ERROR("Call missing mandatory parameter charset.\n");
+        return platch_respond_illegal_arg_std(responseHandle, "Expected `arg['charset'] to be a string.");
+    }
+
+    charset = STDVALUE_AS_STRING(*tmp);
+
+    iconv_t iconv_cd;
+    if ((iconv_cd = iconv_open("UTF-8", charset)) == (iconv_t) -1) {
```

Comment:
similar to above, prefer if conditions without side-effects:
```suggestion
    iconv_t iconv_cd = iconv_open("UTF-8", charset);
    if (iconv_cd == (iconv_t) -1) {
```